microsoft/testsuites/stress/stress_ng_suite.py

```diff
@@ -0,0 +1,161 @@
+        https://manpages.ubuntu.com/manpages/jammy/man1/stress-ng.1.html
+        """,
+        priority=4,
+    )
+    def perf_stress_ng_jobfile(
+        self,
+        variables: Dict[str, Any],
+        environment: Environment,
+        result: TestResult,
+    ) -> None:
+        if self.CONFIG_VARIABLE in variables:
+            jobs = variables[self.CONFIG_VARIABLE]
+            for job_file in jobs:
+                self._run_stress_ng_job(job_file, environment, result)
+        else:
+            raise SkippedException("No jobfile provided for stress-ng")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'cpu' class stressors for 60s each.",
+        priority=4,
+    )
+    def perf_stress_ng_cpu_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "cpu")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'memory' class stressors for 60s each.",
+        priority=4,
+    )
+    def perf_stress_ng_memory_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "memory")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'vm' class stressors for 60s each.",
+        priority=4,
+    )
+    def perf_stress_ng_vm_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "vm")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'io' class stressors for 60s each.",
+        priority=4,
+    )
+    def perf_stress_ng_io_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "io")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'network' class stressors for 60s each.",
+        priority=4,
+    )
+    def perf_stress_ng_network_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "network")
+
+    def _run_stressor_class(self, environment: Environment, class_name: str) -> None:
+        nodes = [cast(RemoteNode, node) for node in environment.nodes.list()]
+        procs: List[Process] = []
+        for node in nodes:
+            procs.append(node.tools[StressNg].launch_class_async(class_name))
+        for proc in procs:
+            proc.wait_result(timeout=self.TIME_OUT, expected_exit_code=0)
+
+    def _run_stress_ng_job(
+        self,
+        job_file: str,
+        environment: Environment,
+        test_result: TestResult,
+    ) -> None:
+        nodes = [cast(RemoteNode, node) for node in environment.nodes.list()]
+        procs: List[Process] = []
+        job_file_name = Path(job_file).name
+        test_status = TestStatus.QUEUED
+        test_msg = ""
+        try:
+            for node in nodes:
+                remote_working_dir = node.working_path / "stress_ng_jobs"
```

Comment:
I think we've been leaning towards using the Path, you can use remote_wroking_dir = node.working_path.join("stress_ng_jobs") and then avoid casting to a path afterwards.

Very excited to have a stress-ng suite! Thank you for this effort 🫡

Reply:
I removed the cast in the later line.  AFAIK, `node.working_path / "stress_ng_jobs"` is same as `node.working_path.joinpath("stress_ng_jobs")`.

Reply:
Actually, the cast to `PurePath` is needed below. Because `job_file` is a string coming from the runbook.